### PR TITLE
Added catch for missing POST body when posted to profileparameters

### LINKED ISSUE #4428: profileparameters POST request with no request body returns Internal Server Error
## I'm submitting a ...

- [X] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

Making a `POST` request to `/profileparameters` with no body in the request returns:
```json
HTTP/1.1 500 Internal Server Error
{
    "alerts": [
        {
            "text": "Internal Server Error",
            "level": "error"
        }
    ]
}
```
## Expected / new behavior:

Most endpoints return:
```json
HTTP/1.1 400 Bad Request 
{
    "alerts": [
        {
            "text": "EOF",
            "level": "error"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by kdamichie**
This PR closes #4428

<hr/>

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Make sure all of the tests pass

For manual testing, try to send a POST to the profileparameters with an empty body.  The result should be a 400 error.

## If this is a bugfix, which Traffic Control versions contained the bug?

-master

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by kdamichie — COMMENTED** [review]
Removed changed traffic-ops-test.conf.  Accidentally commited

File: traffic_ops/testing/api/conf/traffic-ops-test.conf [hunk 1]
```diff
@@ -1,40 +0,0 @@
...
-        "dbname": "traffic_ops_development",
-        "description": "dev database traffic_ops_development",
-        "hostname": "localhost",
-        "password": "twelve12",
-        "port": "5432",
-        "type": "Pg",
-        "user": "traffic_ops"
-    },
-    "noISO": false
-}
```
Comment (ocket8888) [inline traffic_ops/testing/api/conf/traffic-ops-test.conf:40]: I don't think you meant to delete this whole file

**Review by ocket8888 — CHANGES_REQUESTED** [review]

File: traffic_ops/testing/api/v3/profile_parameters_test.go [hunk 2]
```diff
@@ -75,6 +75,11 @@ func TestProfileParameters(t *testing.T) {
 					}},
 					Expectations: utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
 				},
+				"BAD REQUEST when EMPTY BODY PROVIDED": {
+					ClientSession: TOSession,
+					RequestBody:   []tc.ProfileParameter{{}},
+					Expectations:  utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
+				},
```
Comment (ocket8888) [inline traffic_ops/testing/api/v3/profile_parameters_test.go:82]: This doesn't actually test the behavior being fixed in this PR. This a list of length 1 (not 0!) whose only element has the default/"zero" values for all of its properties. [The result when Go marshals that is, accordingly, an array with one object element](https://go.dev/play/p/UadnzuI9JY3).

File: traffic_ops/testing/api/v4/profile_parameters_test.go [hunk 3]
```diff
@@ -76,6 +76,11 @@ func TestProfileParameters(t *testing.T) {
 					},
 					Expectations: utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
 				},
+				"BAD REQUEST when EMPTY BODY PROVIDED": {
+					ClientSession: TOSession,
+					RequestBody:   map[string]interface{}{},
+					Expectations:  utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
+				},
```
Comment (ocket8888) [inline traffic_ops/testing/api/v4/profile_parameters_test.go:83]: This doesn't actually test the behavior being fixed in this PR. [Marshalling an empty map appropriately produces an empty object - not an empty request body](https://go.dev/play/p/A4w_-aQn2HB)!

File: traffic_ops/testing/api/v5/profile_parameters_test.go [hunk 4]
```diff
@@ -76,6 +76,11 @@ func TestProfileParameters(t *testing.T) {
 					},
 					Expectations: utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
 				},
+				"BAD REQUEST when EMPTY BODY PROVIDED": {
+					ClientSession: TOSession,
+					RequestBody:   map[string]interface{}{},
+					Expectations:  utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusBadRequest)),
+				},
```
Comment (ocket8888) [inline traffic_ops/testing/api/v5/profile_parameters_test.go:83]: Same as above

**Review by ocket8888 — COMMENTED** [review]
The test cases here that claim to have empty bodies but actually don't. Note that you can't fix that just by setting `RequestBody` to `nil` - [`nil` marshals to `null`, **not** an empty request body](https://go.dev/play/p/He4OxowBygr). Sending an empty request may not be very easy to do using the same format as all the other tests. I don't think there's anything you can put in for `RequestBody` that will do what you want.

**Comment by ocket8888** [comment]
> I don't think there's anything you can put in for `RequestBody` that will do what you want.

Actually, I take that back. [It's actually fairly easy to create a type that marshals to an empty request body](https://go.dev/play/p/JAf0aV3bZVY).

<details>
<summary>Example</sample>

```go
package main

import (
	"encoding/json"
	"fmt"
)

type emptyJSONTester struct{}

func (emptyJSONTester) MarshalJSON() ([]byte, error) {
	return []byte{}, nil
}

func main() {
	bts, _ := json.Marshal(emptyJSONTester{})
	fmt.Printf("'%s'\n", string(bts))
}
```

</details>

File: traffic_ops/testing/api/v3/profile_parameters_test.go [hunk 5]
```diff
@@ -163,6 +165,16 @@ func TestProfileParameters(t *testing.T) {
 	})
 }
 
+func TestPostWithEmptyBody(t *testing.T) {
+	resp, err := TOSession.Client.Post(TOSession.URL+"/api/5.0/profileparameters", "application/json", nil)
```
Comment (srijeet0406) [inline traffic_ops/testing/api/v3/profile_parameters_test.go:169]: This needs to point to api 3.0

File: traffic_ops/testing/api/v4/profile_parameters_test.go [hunk 6]
```diff
@@ -172,6 +174,16 @@ func TestProfileParameters(t *testing.T) {
 	})
 }
 
+func TestPostWithEmptyBody(t *testing.T) {
+	resp, err := TOSession.Client.Post(TOSession.URL+"/api/5.0/profileparameters", "application/json", nil)
```
Comment (srijeet0406) [inline traffic_ops/testing/api/v4/profile_parameters_test.go:178]: This needs to point to api 4.0

**Review by srijeet0406 — CHANGES_REQUESTED** [review]

File: traffic_ops/testing/api/v3/profile_parameters_test.go (hunk 5 above)
Comment (kdamichie) [reply traffic_ops/testing/api/v3/profile_parameters_test.go]: Correcting for 3.0 and 4.0

**Review by srijeet0406 — APPROVED** [review]
Code looks good. Manual and automated tests pass. Good work! :)

**Review by ocket8888 — APPROVED** [review]

**Pull request merged**